pkg/process/net/common.go

```diff
@@ -379,6 +380,30 @@ func (r *RemoteSysProbeUtil) GetPprof(path string) ([]byte, error) {
 	return io.ReadAll(res.Body)
 }
 
+// GetDiscoveryOpenPorts returns open ports from system-probe.
+func (r *RemoteSysProbeUtil) GetDiscoveryOpenPorts(ctx context.Context) (*discoverymodel.OpenPortsResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, discoveryOpenPortsURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := r.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("got non-success status code: path %s, url: %s, status_code: %d", r.path, discoveryOpenPortsURL, resp.StatusCode)
+	}
+
+	res := &discoverymodel.OpenPortsResponse{}
```

Comment:
maybe out of scope, but I do have a concern about the decoder, it can cause memory explosion